lance-spark-base_2.12/src/main/antlr4/org/apache/spark/sql/catalyst/parser/extensions/LanceSqlExtensions.g4

```diff
@@ -0,0 +1,66 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+grammar LanceSqlExtensions;
+
+singleStatement
+    : statement EOF
+    ;
+
+statement
+    : ALTER TABLE multipartIdentifier ADD COLUMNS columnList FROM identifier        #addColumnsBackfill
```

Comment:
Overall looks good to me, but could we make this a statement instead of forcing an identifier? So I can just do `ALTER TABLE table ADD COLUMNS col1, col2 AS SELECT ... FROM ..` without the need to register the temp view?